data/data/azure/bootstrap/main.tf

```diff
@@ -0,0 +1,156 @@
+    blob  = true
+    queue = false
+    table = false
+    file  = false
+  }
+
+  start  = "${substr(timestamp(), 0, 10)}"
+  expiry = "${substr(timeadd(timestamp(), "24h"), 0, 10)}"
+
+  permissions {
+    read    = true
+    list    = true
+    create  = false
+    add     = false
+    delete  = false
+    process = false
+    write   = false
+    update  = false
+  }
+}
+
+resource "azurerm_storage_container" "ignition" {
+  resource_group_name   = "${var.resource_group_name}"
+  name                  = "ignition"
+  storage_account_name  = "${azurerm_storage_account.ignition.name}"
+  container_access_type = "private"
+}
+
+resource "local_file" "ignition_bootstrap" {
+    content     = "${var.ignition}"
+    filename = "${path.module}/ignition_bootstrap.ign"
+}
+
+resource "azurerm_storage_blob" "ignition" {
+
+  name                    = "bootstrap.ign"
+  source                  = "${local_file.ignition_bootstrap.filename}"
+  resource_group_name     = "${var.resource_group_name}"
+  storage_account_name    = "${azurerm_storage_account.ignition.name}"
+  storage_container_name  = "${azurerm_storage_container.ignition.name}"
+  type = "block"
+}
+
+data "ignition_config" "redirect" {
+  replace {
+    source = "${azurerm_storage_blob.ignition.url}${data.azurerm_storage_account_sas.ignition.sas}"
+  }
+}
+
+data "azurerm_subscription" "primary" {}
+
+resource "azurerm_network_interface" "bootstrap" {
+  name                = "${var.cluster_id}-bootstrap-nic"
+  location            = "${var.region}"
+  resource_group_name = "${var.resource_group_name}"
+
+  ip_configuration {
+    subnet_id                               = "${var.subnet_id}"
+    name                                    = "bootstrap"
+    private_ip_address_allocation           = "Dynamic"
+  }
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "elb_bootstrap" {
+  network_interface_id = "${azurerm_network_interface.bootstrap.id}"
+  backend_address_pool_id = "${var.elb_backend_pool_id}"
+  ip_configuration_name = "bootstrap" #must be the same as nic's ip configuration name.
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "ilb_bootstrap" {
+  network_interface_id = "${azurerm_network_interface.bootstrap.id}"
+  backend_address_pool_id = "${var.ilb_backend_pool_id}"
+  ip_configuration_name = "bootstrap" #must be the same as nic's ip configuration name.
+}
+
+resource "azurerm_virtual_machine" "bootstrap" {
+  name                  = "${var.cluster_id}-bootstrap"
+  location              = "${var.region}"
+  resource_group_name   = "${var.resource_group_name}"
+  network_interface_ids = ["${azurerm_network_interface.bootstrap.id}"]
+  vm_size               = "${var.vm_size}"
+
+  delete_os_disk_on_termination = true
+  storage_os_disk {
+    name              = "bootstraposdisk"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "Premium_LRS"
+    disk_size_gb      = 100
```

Comment:
`var.volume_size` ?

Reply:
I move it up to `azure/main.tf`. Is there a good reason to parameterize this at all, given that the bootstrap node is ephemeral anyway?

Reply:
Hardcoding for bootstrap is totally acceptable.. just drop the variable then.